packages/graph/hash_graph/lib/graph/src/store/mod.rs

```diff
@@ -33,6 +33,9 @@ use crate::{
 /// [`store`]: crate::store
 pub trait Record {
     type Path<'q>: QueryPath;
+    type Metadata;
+
+    fn metadata(&self) -> &Self::Metadata;
```

Comment:
I'm wondering if it's worth a doc-comment explaining why we've done this. I think I'd be pretty confused if I went and looked at the `Entity` definition and saw methods for accessing everything, but then didn't find `metadata`.

Maybe we just have this shadow and call `self.metadata()`?

Reply:
There is a [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#same_name_method) to avoid shadowing of methods, where the method is defined on the trait and on the struct. I don't know if we want to disable the lint, probably it's worth it.

Reply:
Hm, if we have the lint _enabled_ then it's maybe not as bad. Because at least if I went and tried to implement `metadata` (because I incorrectly though it was missing) then clippy would warn me and I'd discover the other one

Reply:
As much as I don't like it, maybe that's actually better, as now we only have one place to refactor if we rename the accessor for example